### PR TITLE
Support PrefixesWithCommunity

### PR DESCRIPTION
**Pull request by oribon**
Support parsing community prefixes

### CODE REVIEW

File: internal/controller/api_to_config.go [hunk 1]
```diff
@@ -96,12 +97,55 @@ func neighborToFRR(n v1beta1.Neighbor, ipv4Prefixes, ipv6Prefixes []string) (*fr
 		case ipfamily.IPv6:
 			res.HasV6Advertisements = true
 		}
-		res.Advertisements = append(res.Advertisements, &frr.AdvertisementConfig{Prefix: p, IPFamily: family})
+
+		// TODO: check that the prefix matches the passed IPv4/IPv6 prefixes
+		advs[p] = &frr.AdvertisementConfig{Prefix: p, IPFamily: family}
+	}
+
+	prefixToCommunities := map[string]sets.Set[string]{}
```
Comment (fedepaol) [inline internal/controller/api_to_config.go:105]: communitiesForPrefix*

File: internal/controller/api_to_config.go [hunk 2]
```diff
@@ -96,12 +97,55 @@ func neighborToFRR(n v1beta1.Neighbor, ipv4Prefixes, ipv6Prefixes []string) (*fr
...
+			prefixToCommunities[p].Insert(pfxs.Community)
+		}
+	}
+
+	for p, c := range prefixToCommunities {
+		adv, ok := advs[p]
+		if !ok { // shouldn't happen as we check in previous loop, just in case
+			return nil, fmt.Errorf("unexpected err - no community prefix matching %s", p)
+		}
+		adv.Communities = sets.List(c)
```
Comment (fedepaol) [inline internal/controller/api_to_config.go:128]: do we have a guaranteed order here?

File: e2etests/tests/advertisement.go [hunk 3]
```diff
@@ -284,6 +284,62 @@ var _ = ginkgo.Describe("Advertisement", func() {
...
+								Prefixes:  []string{"192.168.0.0/24"},
+							},
+							{
+								Community: "10:101",
+								Prefixes:  []string{"192.168.1.0/24"},
+							},
+						}
+					}
+				},
+				validate: func(ppV4 []config.Peer, ppV6 []config.Peer, nodes []v1.Node) {
```
Comment (fedepaol) [inline e2etests/tests/advertisement.go:307]: can you add 
- a test where we have explicit prefixes, only one with community?
- a test where we have large communities
- a test where we have one address with multiple communities

File: internal/controller/api_to_config.go (hunk 1 above)
Comment (oribon) [reply internal/controller/api_to_config.go]: done

File: internal/controller/api_to_config.go (hunk 2 above)
Comment (oribon) [reply internal/controller/api_to_config.go:139]: yes, the List func is explicitly for returning the contents sorted  ( as opposed to the c.UnsortedList() method)

File: e2etests/tests/advertisement.go (hunk 3 above)
Comment (oribon) [reply e2etests/tests/advertisement.go:307]: done, turns out we didn't have the code to support large communities, added a new commit (copy from MetalLB)

**Review by fedepaol — APPROVED** [review]

**Pull request merged**